spec/models/badge_spec.rb

```diff
@@ -234,4 +234,17 @@
       expect(regular_badge.allow_title).to eq(false)
     end
   end
+
+  context "#trigger_badge_granted_event" do
+    it "triggers a 'user_badge_granted' event" do
+      badge = Fabricate(:badge)
+      user = Fabricate(:user)
+      event = DiscourseEvent.track(:user_badge_granted) do
+        badge.trigger_badge_granted_event(user.id)
+      end
+
+      expect(event[:event_name]).to eq(:user_badge_granted)
```

Comment:
This test is redundant since `DiscourseEvent.track` with an event type already [filters](https://github.com/discourse/discourse/blob/c9dab6fd087dd400c904441657f53134393fb298/spec/support/discourse_event_helper.rb#L21) on the event type 😉